### PR TITLE
Add coreos-baremetal to k8s guides

### PR DESCRIPTION
**Pull request by dghubble**
Here is a first crack at adding links to coreos-baremetal...

@robszumski @joshix

### CODE REVIEW

File: Documentation/kubernetes-on-baremetal.md [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+This guide walks a deployer through launching a multi-node Kubernetes cluster on bare metal servers running CoreOS. After completing this guide, a deployer will be able to interact with the Kubernetes API from their workstation using the kubectl CLI tool.
 
-## Deployment Prerequisites
+## Deployment Requirements
 
-### CoreOS Installation
+### CoreOS Version
 
-For all nodes running Kubernetes components (masters & workers), you must use a CoreOS version 962.0.0+ for the `kubelet-wrapper` script to be present in the image. If you wish to use an earlier version (e.g. from the 'stable' channel) see [kubelet-wrapper](kubelet-wrapper.md) for more information.
+All nodes running Kubernetes components (masters & workers) must use a CoreOS version 962.0.0+ for the `kubelet-wrapper` script to be present in the image. If you wish to use an earlier version (e.g. from the 'stable' channel) see [kubelet-wrapper](kubelet-wrapper.md) for more information.
```
Comment (robszumski) [inline Documentation/kubernetes-on-baremetal.md]: Should we say "controllers" here @joshix?

Comment (joshix) [inline Documentation/kubernetes-on-baremetal.md]: Yes. The phrase here should be "controllers and nodes", per http://kubernetes.io/docs/admin/cluster-components/

(The group of all controllers are referred to in one heading in that document as "Master components", but each is a "controller". "Node" is unambiguous.)

File: Documentation/kubernetes-on-baremetal.md [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+
+See the [Kubernetes networking](kubernetes-networking.md) documentation for more information on self-managed networking options.
+
+[coreos-flannel]: https://coreos.com/flannel/docs/latest/flannel-config.html
+[calico-networking]: https://github.com/projectcalico/calico-containers
+[pod-network]: https://github.com/kubernetes/kubernetes/blob/release-1.2/docs/design/networking.md#pod-to-pod
+
+## Automated Provisioning
+
+The [coreos-baremetal](https://github.com/coreos/coreos-baremetal) repository provides guides and a service for network booting (iPXE/GRUB) and provisioning (Ignition) CoreOS clusters. The examples include a [multi-node Kubernetes cluster](https://github.com/coreos/coreos-baremetal/blob/master/Documentation/kubernetes.md) and a [getting started etcd cluster](https://github.com/coreos/coreos-baremetal/blob/master/Documentation/getting-started-rkt.md).
```
Comment (robszumski) [inline Documentation/kubernetes-on-baremetal.md]: I think we can beef this up, to really talk up this tool:

> Booting and installing a large number of CoreOS machines can be automated with a PXE-based workflow. The [coreos-baremetal project](https://github.com/coreos/coreos-baremetal) provides tools and guides for:

> 1. A service for network booting CoreOS with iPXE/GRUB
> 2. A service matching machines to provisioning profiles with Ignition
> 3. Examples for running an [etcd cluster](https://github.com/coreos/coreos-baremetal/blob/master/Documentation/getting-started-rkt.md) and a [multi-node Kubernetes cluster](https://github.com/coreos/coreos-baremetal/blob/master/Documentation/kubernetes.md)**

**Comment by dghubble** [comment]
Better? @robszumski @joshix

File: Documentation/kubernetes-on-baremetal.md [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
+[pod-network]: https://github.com/kubernetes/kubernetes/blob/release-1.2/docs/design/networking.md#pod-to-pod
+
+## Automated Provisioning
+
+Booting and provisioning CoreOS clusters can be automated using the [coreos-baremetal](https://github.com/coreos/coreos-baremetal) project. It includes:
+
+* Guides for configuring an network boot environment with iPXE/GRUB
+* An HTTP/gRPC service for booting and provisioning machines. Match machines by their hardware attributes and serve templated [Ignition](https://coreos.com/ignition/docs/latest/c) configs or cloud-configs.
+* [Example](https://github.com/coreos/coreos-baremetal/tree/master/examples) clusters including an [etcd cluster](https://github.com/coreos/coreos-baremetal/blob/master/Documentation/getting-started-rkt.md), [multi-node Kubernetes cluster](https://github.com/coreos/coreos-baremetal/blob/master/Documentation/kubernetes.md), and [self-hosted](https://github.com/coreos/coreos-baremetal/blob/master/Documentation/bootkube.md) Kubernetes cluster. 
+
```
Comment (robszumski) [inline Documentation/kubernetes-on-baremetal.md]: Much improved. Only issue I can spot, do we need to have a main call to action here? Like "Get started with automated provisioning" or similar. Maybe even unlink a few of the examples so we can funnel you into a single spot.

File: Documentation/kubernetes-on-baremetal.md [hunk 4]
```diff
@@ -1,13 +1,34 @@
 # Kubernetes Installation on Bare Metal &amp; CoreOS
 
-This guide walks a deployer through launching a multi-node Kubernetes cluster on bare metal servers running CoreOS.
-After completing this guide, a deployer will be able to interact with the Kubernetes API from their workstation using the kubectl CLI tool.
+This guide walks a deployer through launching a multi-node Kubernetes cluster on bare metal servers running CoreOS. After completing this guide, a deployer will be able to interact with the Kubernetes API from their workstation using the kubectl CLI tool.
 
-## Deployment Prerequisites
+## Deployment Requirements
```
Comment (joshix) [inline Documentation/kubernetes-on-baremetal.md]: Sentence case headings -- `## Deployment requirements`

File: Documentation/kubernetes-on-baremetal.md [hunk 5]
```diff
@@ -1,13 +1,34 @@
 # Kubernetes Installation on Bare Metal &amp; CoreOS
 
-This guide walks a deployer through launching a multi-node Kubernetes cluster on bare metal servers running CoreOS.
-After completing this guide, a deployer will be able to interact with the Kubernetes API from their workstation using the kubectl CLI tool.
+This guide walks a deployer through launching a multi-node Kubernetes cluster on bare metal servers running CoreOS. After completing this guide, a deployer will be able to interact with the Kubernetes API from their workstation using the kubectl CLI tool.
```
Comment (joshix) [inline Documentation/kubernetes-on-baremetal.md]: s/kubectl/`kubectl`/

File: Documentation/kubernetes-on-baremetal.md [hunk 6]
```diff
@@ -1,13 +1,34 @@
...
 
-## Deployment Prerequisites
+## Deployment Requirements
 
-### CoreOS Installation
+### CoreOS Version
 
-For all nodes running Kubernetes components (masters & workers), you must use a CoreOS version 962.0.0+ for the `kubelet-wrapper` script to be present in the image. If you wish to use an earlier version (e.g. from the 'stable' channel) see [kubelet-wrapper](kubelet-wrapper.md) for more information.
+All Kubernetes controllers and nodes must use a CoreOS version 962.0.0+ for the `kubelet-wrapper` script to be present in the image. If you wish to use an earlier version (e.g. from the 'stable' channel) see [kubelet-wrapper](kubelet-wrapper.md) for more information.
```
Comment (joshix) [inline Documentation/kubernetes-on-baremetal.md]: "...must use CoreOS version 962 or greater..."

File: Documentation/kubernetes-on-baremetal.md [hunk 7]
```diff
@@ -1,13 +1,34 @@
...
+
+### Kubernetes Pod Network
+
+These guides use [flannel][coreos-flannel] as a software-defined overlay network to manage routing of the [pod network][pod-network]. However, bare metal is a common platform where a self-managed network may be used, due to the flexibility provided by physical networking gear. One common alternative is to use [Calico][calico-networking] networking, avoiding the use of an overlay and allowing interop with physical networking gear using BGP.
```
Comment (joshix) [inline Documentation/kubernetes-on-baremetal.md]: This configuration uses the [flannel] overlay network to manage the [pod network]. Many bare metal configurations may instead have an existing self-managed network. In this scenario, it is common to use [Calico] to manage pod network policy while omitting the overlay network, and interoperating with existing physical network gear over BGP.

File: Documentation/kubernetes-on-baremetal.md (hunk 4 above)
Comment (dghubble) [inline Documentation/kubernetes-on-baremetal.md]: Why this specific one and not the other `##` headings? Doesn't it look better with capitalized headings? I haven't seen folks doing it this way and I'm curious.

Comment (joshix) [inline Documentation/kubernetes-on-baremetal.md]: I was not explicit, but intended it to be applied to all headings. See https://github.com/coreos/docs/blob/master/STYLE.md#headings

This has been contentious, but sentence case is both more modern and simpler to write without inciting debate about what word is of Title Consequence or Not. We still selectively employ Title Case in some visual design elements but the general rule for a general document is Sentence case.

Comment (dghubble) [inline Documentation/kubernetes-on-baremetal.md]: Updated

**Comment by joshix** [comment]
This doesn't quite become the single funnel @robszumski waffled about a little, but I think it LGTM for purpose.

**Comment by robszumski** [comment]
LGTM

**Pull request merged**